blob-parser/__init__.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import os, csv, json, traceback
+from time import time
+import xxhash
+
+from bs4 import BeautifulSoup
+
+import azure.functions as func
+from azure.storage.blob import BlobServiceClient, ContainerClient
+
+from shared import pre2017, post2017
+from shared.helpers import *
+
+def main(myblob: func.InputStream):
+    logging.info(f"Python blob trigger function processed blob \n"
+                 f"Name: {myblob.name}\n"
+                 f"Blob Size: {myblob.length} bytes")
+
+    # Get case info from file name, which looks like: case-html/15-1367CR-3:hays:12_13_2022:96316e53a9b706e0.html
+    # First strip off case-html/ from beginning and .html from end of blob name
+    stripped_name = myblob.name.strip("case-html/.")
+    # Then split by : as delimiter
+    file_info = stripped_name.split(":")
+    case_num = file_info[0]
+    county = file_info[1]
+    case_date = file_info[2]
+    file_hash = file_info[3][:-5]
+    logging.info(f"Retrieved the following metadata: \n"
+                f"Case Date: {case_num}\n"
+                f"County: {county}\n"
+                f"Date Scraped: {case_date}\n"
+                f"File Hash: {file_hash}")
+    
+    # get county version year information to determine which parser to use
+    base_url = odyssey_version = None
+    with open(
+        os.path.join(
+            os.path.dirname(__file__), "..", "resources", "texas_county_data.csv"
+        ),
+        mode="r",
+    ) as file_handle:
+        csv_file = csv.DictReader(file_handle)
+        for row in csv_file:
+            if row["county"].lower() == county.lower():
+                odyssey_version = int(row["version"].split(".")[0])
+                break
+    if not odyssey_version:
+        raise Exception(
+            "The required data to scrape this county is not in ./resources/texas_county_data.csv"
+        )
+
+    # call parser
+    START_TIME = time()     
+    logging.info(f"Processing {case_num} - {county} with {odyssey_version} Odyssey parser...")
+    try:
+        case_soup = BeautifulSoup(myblob, "html.parser", from_encoding="UTF-8")
+
+        if odyssey_version < 2017:
+            case_data = pre2017.parse(case_soup, case_num)
+            logging.info("Pre 2017")
+        else:
+            logging.info("Post 2017")
+
+        # initialize blob container client for sending json files to
+        blob_connection_str = os.getenv("AzureWebJobsStorage")
+        container_name_json = os.getenv("blob_container_name_json")
+        blob_service_client: BlobServiceClient = BlobServiceClient.from_connection_string(
+            blob_connection_str
+        )
+        container_client = blob_service_client.get_container_client(container_name_json)
+
+        # Write JSON data
+        file_hash = xxhash.xxh64(json.dumps(case_data)).hexdigest()

```

Comment:
Let's use the file_hash from the .html files rather than rehashing the .json files, so it's clear which htmls are linked to which jsons